config/identifiable.go

```diff
@@ -78,6 +91,12 @@ func (id ComponentID) Name() string {
 	return id.nameVal
 }
 
+// WithName returns a new ComponentID, with the same type as the current instance,
+// and the given name.
+func (id ComponentID) WithName(nameVal string) ComponentID {
```

Comment:
The usage of `WithName` in the code looks a bit unclear, e.g. `defaultID.WithName` is somewhat weird. How about simply having a `NewIDWthName` func?

Reply:
That will make us to pass type + name. I can try to see how it looks in couple of cases and we can decide

Reply:
Done.